src/pages/index.tsx

```diff
@@ -1,15 +1,39 @@
-import { Button } from "@/components/ui/button";
-import { signOut, useSession } from "next-auth/react";
-import React from "react";
+import { Footer } from "@/components/Navigation/Footer";
+import Header from "@/components/Navigation/Header";
+import { SessionProvider } from "next-auth/react";
+import React, { useEffect, useState } from "react";
+import { useSession } from "next-auth/react";
+import { userDataSchema } from "@/components/ProfileModal/ProfileModal";
+import { z } from "zod";
 
 const Home = () => {
-  const { data, status } = useSession();
+  const { data: session } = useSession();
+  const currentUser = session?.user;
+  const [userData, setUserData] = useState<z.infer<typeof userDataSchema>>();
+
+  useEffect(() => {
+    if (currentUser) {
+      getUserData();
+    }
+  }, [currentUser]);
+
+  async function getUserData() {
+    try {
+      const response = await fetch(`/api/users/${currentUser?._id}`);
+      const data = await response.json();
+      setUserData(data.data);
```

Comment:
I was testing the Header and logging in wasn't changing the status for me, I believe it's cuz this should be setUserData(data) instead?

Reply:
Yup, you are so right. The implementation changed when I pulled from main and I should have checked after merging from main. Sorry about that!